tasks/sst_metrics_utils/matcher.py

```diff
@@ -32,6 +32,8 @@ def match_catalogs(inputs, photoCalibs, vIds, matchRadius, logger=None):
                                        'PSF Ellipticity 1'))
     mapper.addOutputField(Field[float]('psf_e2',
                                        'PSF Ellipticity 1'))
+    mapper.addOutputField(Field[np.int32]('filt',
```

Comment:
Ah.  I was assuming you were adding the filter to the visit id mapping that's passed to `MultiMatch`.  If you are explicitly adding it to the matched catalog, is there a reason not to make this a string field to avoid the issues you brought up before about mapping back and forth between integers and strings?

Reply:
I think I tried that but was having the same issue that strings require a size. I'll try it again to see if I can make it work. (Would prefer to avoid the mapping if possible!)

Reply:
I'm still getting errors when trying to implement it as `mapper.addOutputField(Field[str]('filter', 'filter name'))`. (I also tried "String" and np.str, with no success. Unfortunately from the traceback it's unclear whether the error is in the `mapper` part or later when I assign to the array using `tmpCat['filter'] = np.repeat(vId['abstract_filter'], len(oldSrc))`.

Reply:
You need to give it a `size` e.g. https://github.com/lsst/afw/blob/master/tests/test_schema.py#L56

Does that help?

Reply:
No, I get a different error (a "TypeError"). Should I keep hammering away at this to sort it out, or just keep the dictionary for now?

Reply:
I guess just do what works for now.

Reply:
OK - good to merge (once I resolve the conflicts)? [We can fix the dictionary thing later.]